R/download.R

```diff
@@ -4,16 +4,24 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
-#' Download Bugzilla issues and comments using Perceval
+#' Download Bugzilla issues and comments using Perceval.
+#'
+#' Note that for the Bugzilla REST API backend, Bugzilla sites may limit the number of bugs that can be retrieved at one time.
+#' Thus, the max_bugs parameter needs to be set correctly to ensure all bugs are retrieved and that
+#' the json data is not broken. If you get an error trying to parse the data downloaded with this
 #'
 #' @param perceval_path path to perceval binary
 #' @param bugzilla_site link to specific bugzilla site
 #' @param datetime fetch bugs updated since this date (in any ISO 8601 format, e.g., 'YYYY-MM-DD HH:mm:SS+|-HH:MM'))
 #' @param backend either traditional "bugzilla" or "bugzillarest" for Bugzilla >= 5.0 servers
+#' @param max_bugs the maximum number of bugs requested on the same query. This acts as the limit parameter
+#' in the Bugzilla REST API. Bugzilla sites may have specific limits set, so make sure to change the max_bugs
+#' parameter accordingly to correctly download the data when using the "bugzillarest" backend.
 #' @export
-download_bugzilla <- function(perceval_path, bugzilla_site, datetime, backend="bugzilla"){
+download_bugzilla <- function(perceval_path, bugzilla_site, datetime, backend="bugzilla", max_bugs=500){
```

Comment:
Malia was having issues using my download_bugzilla for Redhat Bugzilla using the bugzillarest backend, and I also noticed problems, so I tried it using max_bugs set to 20 and it worked. Thus, this is the same case as the bugzilla crawler issue with the pagination. I've added the "max_bugs" parameter which acts as the "limit" parameter so that users can adjust this accordingly.

Reply:
Just to be clear, when you say the same issue:

You are saying Perceval will defer to the user to make a best guess on the page limit, and if they don't have any idea what that is, it would skip from page 0-20 to 500-520, hence skipping issues on Red Hat case?

And, I see you made one change for the parameter here, but does that mean we also need to add that file parsing check to adjust the limit so it doesn't skip?

Reply:
Hi, yes that seems to be the case. When I change the max_bugs parameter, I noticed a different number of issues being returned.

>And, I see you made one change for the parameter here, but does that mean we also need to add that file parsing check to adjust the limit so it doesn't skip?

Do you mean saving each of the files from the Perceval calls into a folder and checking how many bugs are in the first one? Yes, I think we would need that check to adjust the limit. Should I save the data similar to how I did it for the crawler downloader then?